js/network.js

```diff
@@ -27,7 +27,21 @@ async function sign_tx(api, settings){
   var xrp_secret = document.getElementById("xrp_secret");
   var xrp_address = document.getElementById("xrp_address");
   var eth_address = document.getElementById("eth_address")
-
+  var keyPair = {}
+
+  // generate the keyPair to use for signing purposes, either from the private key, or the secret, depending on what was specified
+  if (offline_api.isValidSecret(xrp_secret.value)) {
+    // a secret key was specified
+    keyPair = offline_api.deriveKeypair(xrp_secret.value);
+  } else {
+    // a private key was specified (64 or 66 chars, double-zeroed or not)
+    const public64 = (xrp_secret.length > 64) ? xrp_secret.value.slice(2) : xrp_secret.value;
```

Comment:
Again this can be simplified here by assuming keys are 66 chars in length. If you know a reason why we shouldn't simplify, please let me know as perhaps I am missing something.